main/src/com/google/refine/exporters/sql/SqlCreateBuilder.java

```diff
@@ -0,0 +1,145 @@
+        }
+        StringBuffer createSB = new StringBuffer();
+
+        JSONArray columnOptionArray = options == null ? null : JSONUtilities.getArray(options, "columns");
+        
+        final boolean trimColNames = options == null ? false : JSONUtilities.getBoolean(options, "trimColumnNames", false);
+      
+        int count = columnOptionArray.length();
+
+        for (int i = 0; i < count; i++) {
+            JSONObject columnOptions = JSONUtilities.getObjectElement(columnOptionArray, i);
+            if (columnOptions != null) {
+                String name = JSONUtilities.getString(columnOptions, "name", null);
+                String type = JSONUtilities.getString(columnOptions, "type", "VARCHAR");
+                String size = JSONUtilities.getString(columnOptions, "size", "");
+                //logger.info("Before Trim Column Names::" + name);
+                
+                if (name != null) {
+                    if(trimColNames) {
+                        String trimmedCol = name.replaceAll("\\s", "");
+                        createSB.append( trimmedCol + " ");
+                        //logger.info("After Trim Column Names::" + name);
+                    }else{
+                        createSB.append(name + " ");
+                    }
+                   
+                    if (type.equals("VARCHAR")) {
+                        if (size.isEmpty()) {
+                            size = "255";
+                        }
+                        createSB.append(type + "(" + size + ")");
+
+                    } else if (type.equals("CHAR")) {
+                        if (size.isEmpty()) {
+                            size = "10";
+                        }
+                        createSB.append(type + "(" + size + ")");
+
+                    } else if (type.equals("INT") || type.equals("INTEGER")) {
+                        if (size.isEmpty()) {
+                            createSB.append(type);
+                        } else {
+                            createSB.append(type + "(" + size + ")");
+                        }
+
+                    } else if (type.equals("NUMERIC")) {
+                        if (size.isEmpty()) {
+                            createSB.append(type);
+                        } else {
+                            createSB.append(type + "(" + size + ")");
+                        }
+                    } else {
+                        createSB.append(type);
+                    }
+
+                    if (i < count - 1) {
+                        createSB.append(",");
+                    }
+                    createSB.append("\n");
+                }
+            }
+        }
+
+        StringBuffer sql = new StringBuffer();
+
+        boolean includeDrop = JSONUtilities.getBoolean(options, "includeDropStatement", false);
+        if (includeDrop) {
+            sql.append("DROP TABLE " + table + ";\n");
+        }
+
+        sql.append("CREATE TABLE ").append(table);
```

Comment:
Not sure is there any database type required here? Oracle or Mysql?

Reply:
This should work for MySQL, Oracle, PgSql and MariaDB